### PR TITLE
Handle search within team

### PR DESCRIPTION
**Pull request by tiago-loureiro**
This solves the problem for on premise deployments where it is not desirable have team members connecting outside their team

### CODE REVIEW

**Comment by tiago-loureiro** [comment]
> This only touches `/users/handles/<handle>` but it doesn't change the logic of `/search/contacts?q=<prefix>`, meaning users still can connect to users outside their team (team users if they set their searchable status to true, and to users not part of a team).

Right, although there's no UI for this (setting status to true) and typically admins ensure that no non-team accounts can be created. However, your point is still valid, I will touch the other endpoint as well, thanks @jschaul !

> Please also link to an issue describing the problem that this PR attempts to solve.

Sure.

**Comment by tiago-loureiro** [comment]
I forgot to comment again, ready for re-review.

File: services/brig/test/integration/API/Search.hs [hunk 1]
```diff
@@ -41,7 +41,7 @@ testOptInOut brig = do
     refreshIndex brig
     assertSearchable "opted in" brig uid1 True
     assertCanFind brig uid2 uid1 h1
-
+    -- Even if users opt in, team users in other teams should not be found
```
Comment (jschaul) [inline services/brig/test/integration/API/Search.hs]: Did you mean to write a few more test lines here? Or what does this comment relate to? (maybe move to the test case that does this / delete this line)

Comment (tiago-loureiro) [reply services/brig/test/integration/API/Search.hs]: Ah please ignore this, it was moved to `testNonSearchableAcrossTeam`

**Pull request merged**